qemu/tests/check_coredump.py

```diff
@@ -148,51 +148,69 @@ def run(test, params, env):
     4) If yes, copy them to working directory.
     """
 
+    error_msgbox = []
+    error.context("Check crash/coredump in test result dir", logging.info)
+    root_dir = os.path.dirname(os.path.dirname(test.debugdir))
+    crash_dir = []
+    for prefix in ["crash", "core"]:
+        pattern = "%s/*/%s.*" % (root_dir, prefix)
+        crash_dir += glob.glob(pattern)
+    if crash_dir:
+        crash_msg = ("Crash/Coredump happened, " +
+                     "Please check dump files under %s" % crash_dir)
+        error_msgbox.append(crash_msg)
+
     # Preliminary
     # yum install libguestfs libguestfs-tools libguestfs-winsupport
     try:
         os_dep.command("guestmount")
-    except:
-        warn_msg = "Need packages: libguestfs libguestfs-tools" + \
-                   " libguestfs-winsupport"
+        # define the file name need to be checked
+        file_check_win_default = "Windows/dump"
+        file_check_linux_default = "var/crash/*"
+        host_mountpoint_default = "mnt/mountpoint"
+
+        host_mountpoint = params.get(
+            "host_mountpoint",
+            host_mountpoint_default)
+        host_mountpoint = utils_misc.get_path(test.debugdir, host_mountpoint)
+        file_chk_for_win = params.get(
+            "coredump_check_win",
+            file_check_win_default)
+        file_chk_for_linux = params.get("coredump_check_linux",
+                                        file_check_linux_default)
+
+        # check if the host_mountpoint exists.
+        if not (os.path.isdir(host_mountpoint) and
+                os.path.exists(host_mountpoint)):
+            os.makedirs(host_mountpoint)
+
+        coredump_file_exists = False
+        check_files = [file_chk_for_win, file_chk_for_linux]
+        check_results = []
+
+        error.context("Get all the images name", logging.info)
+        images = get_images()
+        error.context("images: %s" % images, logging.info)
+
+        # find all the images
+        # mount per-image to check if the dump file exists
+        error.context("Check coredump file per-image", logging.info)
+        for image in images:
+            status, chk_msgs = check_images_coredump(image,
+                                                     host_mountpoint,
+                                                     check_files, test.debugdir)
+            coredump_file_exists = coredump_file_exists or status
+            if status:
+                check_results.append((image, chk_msgs))
+
```

Comment:
context: Could not mount guest image /root/autotest-devel/client/tests/virt/shared/data/images/rhel71-64-virtio.qcow2.  (there is "." afte qcow2)